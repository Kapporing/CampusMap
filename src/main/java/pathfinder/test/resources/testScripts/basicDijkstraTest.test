# Basic Dijkstra on a simple graph
CreateGraph sanity_check

AddNode sanity_check A
AddNode sanity_check B
AddNode sanity_check C
AddNode sanity_check D
AddNode sanity_check E
AddNode sanity_check F
AddNode sanity_check G

AddEdge sanity_check A B 2.0
AddEdge sanity_check A C 1.0

AddEdge sanity_check B C 5.0
AddEdge sanity_check B D 11.0
AddEdge sanity_check B E 3.0

AddEdge sanity_check C G 15.0

AddEdge sanity_check D E 2.0

AddEdge sanity_check E C 1.0
AddEdge sanity_check E G 4.0
AddEdge sanity_check E F 5.0

AddEdge sanity_check F D 1.0
AddEdge sanity_check F G 1.0

FindPath sanity_check A F
FindPath sanity_check A A
FindPath sanity_check F A
