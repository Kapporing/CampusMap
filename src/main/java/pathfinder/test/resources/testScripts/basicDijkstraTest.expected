# Basic Dijkstra on a simple graph
created graph sanity_check

added node A to sanity_check
added node B to sanity_check
added node C to sanity_check
added node D to sanity_check
added node E to sanity_check
added node F to sanity_check
added node G to sanity_check

added edge 2.000 from A to B in sanity_check
added edge 1.000 from A to C in sanity_check

added edge 5.000 from B to C in sanity_check
added edge 11.000 from B to D in sanity_check
added edge 3.000 from B to E in sanity_check

added edge 15.000 from C to G in sanity_check

added edge 2.000 from D to E in sanity_check

added edge 1.000 from E to C in sanity_check
added edge 4.000 from E to G in sanity_check
added edge 5.000 from E to F in sanity_check

added edge 1.000 from F to D in sanity_check
added edge 1.000 from F to G in sanity_check

path from A to F:
A to B with weight 2.000
B to E with weight 3.000
E to F with weight 5.000
total cost: 10.000
path from A to A:
total cost: 0.000
path from F to A:
no path found
